app/src/main/java/ch/hikemate/app/model/route/HikeRoutesRepositoryOverpass.kt

```diff
@@ -40,7 +57,14 @@ class HikeRoutesRepositoryOverpass(val client: OkHttpClient) : HikeRoutesReposit
 
     setRequestHeaders(requestBuilder)
 
-    client.newCall(requestBuilder.build()).enqueue(OverpassResponseHandler(onSuccess, onFailure))
+    // The callback updates the cache
+    val onSuccessWithCache = { routes: List<HikeRoute> ->
+      cachedHikeRoutes[bounds] = routes
+      onSuccess(routes)
+    }
+    client
+        .newCall(requestBuilder.build())
+        .enqueue(OverpassResponseHandler(onSuccessWithCache, onFailure))
```

Comment:
Correction: The OverpassResponseHandler is an inner class, so the cache update should be implemented there instead of hacking the onSuccess trigger

Reply:
I changed this tell me if it's what you meant.